docs/source/start/declarative.rst

```diff
@@ -0,0 +1,134 @@
+..
+   SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+   SPDX-License-Identifier: Apache-2.0
+
+.. _declarative-modelling:
+
+*********************
+Declarative modelling
+*********************
+
+Since v0.5, py-capellambse supports declarative modelling with the
+``capellambse.decl`` module.
+
+The YAML-based declarative modelling engine combines a few simple concepts into
+a powerful, but easy to use file format. These files can then be applied to any
+model supported by py-capellambse.
+
+Example
+=======
+
+Here is an example YAML file that declares a simple coffee machine with a
+couple of functions, and functional exchanges between them:
+
+.. literalinclude:: ../../../tests/data/decl/coffee-machine.yml
+   :language: yaml
+   :lines: 4-
+   :lineno-start: 1
+   :linenos:
+
+Assuming this file is saved as ``coffee-machine.yml``, it can then be applied
+to a locally stored Capella model with a command like this:
+
+.. code-block:: sh
+
+   python -m capellambse.decl --model path/to/model.aird coffee-machine.yml
+
+Refer to the :py:func:`capellambse.cli_helpers.loadcli` documentation to find
+out the supported argument format for ``--model``.
+
+Declarative YAML can also be applied programmatically, by calling the
+:py:func:`capellambse.decl.apply` function. It takes a (loaded) CapellaMBSE
+model, and either a path to a file or a file-like object. To pass in a string
+containing YAML, wrap it in ``io.StringIO``:
+
+.. code-block:: python
+   :emphasize-lines: 5
+
+   import io, capellambse.decl
+   my_model = capellambse.MelodyModel(...)
+   my_yaml = "..."
+
+   capellambse.decl.apply(my_model, io.StringIO(my_yaml))
+
+   my_model.save()
+
+Format description
+==================
+
+The expected YAML follows a simple format, where a parent object (i.e. an
+object that already exists in the model) is selected, and one or more of three
+different operations is applied to it: ``create``-ing new child objects,
+``modify``-ing the object itself, or ``delete``-ing one or more children.
```

Comment:
> Can you explain why you made `_OPERATIONS` and its values `_operate_create`, `_operate_modify` and `_operate_delete` private?

They're private because the very fact that they exist is an implementation detail – they're not supposed to be called directly, and therefore have no reason to show up in the documentation of the public API.

If there's interesting documentation about the behavior of private functions/methods, it should be placed on the public functions that call them, i.e. in this case `apply`.

Reply:
> What do you think about converting this enumeration of available operations to an unordered list:

Another great idea! Added that as well. :)